sk0-front/src/app/common/services/registerEmailValidator.ts

```diff
@@ -0,0 +1,26 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from "@angular/forms";
+import { Observable, of } from "rxjs";
+
+export function RegisterEmailValidator(): ValidatorFn {
+  return (control: AbstractControl): ValidationErrors | null => {
+    if (!control.value) {
+      return null;
+    }
+
+    const [username, serverDomain] = control.value.split("@");
+
+    if (!username || !serverDomain) {
+      return null;
+    }
+
+    const [server, domain] = serverDomain.split(".");
+
+    const nameNoSpecChar = /^[a-zA-Z0-9._%+-]+$/.test(username);
+
+    const domainExists = !!domain;
+
+    const emailValid = nameNoSpecChar && domainExists;
+
+    return !emailValid ? { invalidEmail: true } : null;
+  };
+}
```

Comment:
This file is not really a service, I think you might wanted to create another package with `utils/validators` or what the name that you would like. And probably you should unit test it, probably we might wanted to use this function at different places as well with different context.